File: week1/problem_set_1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_bricks(int height);


int main(void)
{
    // Get height of pyramid
    int height = get_height();

    // Print pyramid of bricks
    print_bricks(height);

}

// Function prompt and get user input
int get_height(void)
{
    // Gets an integer for height, loop back if not between 1 and 8 inclusive
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    return n;
}

// Function to print bricks of each row
void print_bricks(int height)
{
    // Loops through each row
    for (int i = 1; i <= height; i++)
    {
        // Loops through each character in a row, determines to print a hash or a space
        for (int j = 1; j <= height; j++)
        {
            if (j > height - i)
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }
        printf("\n");
    }
}

File: week6_Python/problem_set_6/sentimental-mario-less/mario.py
# TODO

# Recreation of mario.c in problem set 1
from cs50 import get_int

# Get user input for height
height = -1
while (height < 1 or height > 8):
    height = get_int("Height: ")

# Loops through each row
for i in range(1, height + 1):
    for j in range(1, height + 1):
        # Loops through each character in a row, determines to print a hash or a space
        if j > height - i:
            print("#", end="")
        else:
            print(" ", end="")
    print("")


